File: templates/kmp/screens/onboarding/view/OnboardingScreen.kt
package features.onboarding.view

import androidx.compose.foundation.layout.*
import androidx.compose.foundation.pager.HorizontalPager
import androidx.compose.foundation.pager.rememberPagerState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import core.components.AppButton
import core.components.AppGhostButton
import core.theme.AppColors
import kotlinx.coroutines.launch

private data class Slide(val emoji: String, val title: String, val desc: String)

private val slides = listOf(
    Slide("🚀", "Welcome", "Discover a new way to manage your life."),
    Slide("🔒", "Secure & Private", "Your data is encrypted and protected."),
    Slide("⚡", "Fast & Reliable", "Lightning-fast performance."),
    Slide("🎯", "Get Started", "Create your account today!"),
)

@Composable
fun OnboardingScreen(onFinish: () -> Unit = {}) {
    val pagerState = rememberPagerState(pageCount = { slides.size })
    val scope = rememberCoroutineScope()

    Column(modifier = Modifier.fillMaxSize()) {
        if (pagerState.currentPage < slides.size - 1) {
            Box(Modifier.fillMaxWidth().padding(16.dp), contentAlignment = Alignment.CenterEnd) {
                AppGhostButton("Skip", onClick = onFinish)
            }
        }

        HorizontalPager(state = pagerState, modifier = Modifier.weight(1f)) { page ->
            Column(
                modifier = Modifier.fillMaxSize().padding(32.dp),
                verticalArrangement = Arrangement.Center,
                horizontalAlignment = Alignment.CenterHorizontally,
            ) {
                Text(slides[page].emoji, style = MaterialTheme.typography.displayLarge)
                Spacer(Modifier.height(32.dp))
                Text(slides[page].title, style = MaterialTheme.typography.headlineLarge, color = AppColors.TextPrimary)
                Spacer(Modifier.height(16.dp))
                Text(slides[page].desc, style = MaterialTheme.typography.bodyMedium, color = AppColors.TextSecondary)
            }
        }

        // Dots
        Row(Modifier.fillMaxWidth().padding(vertical = 16.dp), horizontalArrangement = Arrangement.Center) {
            slides.indices.forEach { i ->
                Surface(
                    modifier = Modifier.padding(horizontal = 4.dp).size(if (pagerState.currentPage == i) 24.dp else 8.dp, 8.dp),
                    shape = RoundedCornerShape(4.dp),
                    color = if (pagerState.currentPage == i) AppColors.Primary else AppColors.SurfaceLight,
                ) {}
            }
        }

        Box(Modifier.padding(24.dp)) {
            AppButton(
                text = if (pagerState.currentPage == slides.size - 1) "Get Started" else "Next",
                onClick = {
                    if (pagerState.currentPage < slides.size - 1) {
                        scope.launch { pagerState.animateScrollToPage(pagerState.currentPage + 1) }
                    } else onFinish()
                },
            )
        }
    }
}

File: templates/flutter/screens/onboarding/presentation/onboarding_screen.dart
import 'package:flutter/material.dart';
import 'package:go_router/go_router.dart';
import '../../../core/theme/app_colors.dart';
import '../../../core/widgets/widgets.dart';

class OnboardingScreen extends StatefulWidget {
  const OnboardingScreen({super.key});
  @override
  State<OnboardingScreen> createState() => _OnboardingScreenState();
}

class _OnboardingScreenState extends State<OnboardingScreen> {
  final _pageController = PageController();
  int _currentPage = 0;

  final _slides = const [
    {'emoji': '🚀', 'title': 'Welcome', 'desc': 'Discover a new way to manage your life.'},
    {'emoji': '🔒', 'title': 'Secure & Private', 'desc': 'Your data is encrypted and protected.'},
    {'emoji': '⚡', 'title': 'Fast & Reliable', 'desc': 'Lightning-fast performance.'},
    {'emoji': '🎯', 'title': 'Get Started', 'desc': 'Create your account today!'},
  ];

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: SafeArea(
        child: Column(
          children: [
            if (_currentPage < _slides.length - 1)
              Align(
                alignment: Alignment.topRight,
                child: TextButton(onPressed: () => context.go('/signin'), child: const Text('Skip')),
              ),
            Expanded(
              child: PageView.builder(
                controller: _pageController,
                onPageChanged: (i) => setState(() => _currentPage = i),
                itemCount: _slides.length,
                itemBuilder: (context, index) {
                  final slide = _slides[index];
                  return Padding(
                    padding: const EdgeInsets.all(32),
                    child: Column(
                      mainAxisAlignment: MainAxisAlignment.center,
                      children: [
                        Text(slide['emoji']!, style: const TextStyle(fontSize: 80)),
                        const SizedBox(height: 32),
                        Text(slide['title']!, style: Theme.of(context).textTheme.headlineLarge, textAlign: TextAlign.center),
                        const SizedBox(height: 16),
                        Text(slide['desc']!, style: Theme.of(context).textTheme.bodyMedium, textAlign: TextAlign.center),
                      ],
                    ),
                  );
                },
              ),
            ),
            Row(
              mainAxisAlignment: MainAxisAlignment.center,
              children: List.generate(_slides.length, (i) => AnimatedContainer(
                duration: const Duration(milliseconds: 300),
                margin: const EdgeInsets.symmetric(horizontal: 4),
                width: _currentPage == i ? 24 : 8,
                height: 8,
                decoration: BoxDecoration(
                  color: _currentPage == i ? AppColors.primary : AppColors.surfaceLight,
                  borderRadius: BorderRadius.circular(4),
                ),
              )),
            ),
            Padding(
              padding: const EdgeInsets.all(24),
              child: AppButton(
                title: _currentPage == _slides.length - 1 ? 'Get Started' : 'Next',
                onPressed: () {
                  if (_currentPage < _slides.length - 1) {
                    _pageController.nextPage(duration: const Duration(milliseconds: 300), curve: Curves.ease);
                  } else {
                    context.go('/signin');
                  }
                },
              ),
            ),
          ],
        ),
      ),
    );
  }
}
